rusoto/credential/src/instance_metadata.rs

```diff
@@ -84,12 +92,17 @@ impl Future for InstanceMetadataProviderFuture {
     fn poll(&mut self) -> Poll<Self::Item, Self::Error> {
         let new_state = match self.state {
             InstanceMetadataFutureState::Start => {
-                let new_future = get_role_name(&self.client, self.timeout)?;
+                let new_future = get_role_name(&self.client, self.timeout, self.port_for_testing.clone())?;
```

Comment:
Bunch of `clones` aren't great. 🤷‍♀

Reply:
It's not, but in the usual path this is only cloning None. I don't know if that allocates. (Maybe `Option::cloned` is more correct here?)